File: pdanet/src/main/java/net/artux/pdanetwork/service/user/reset/ResetService.java
package net.artux.pdanetwork.service.user.reset;

import net.artux.pdanetwork.models.Status;
import net.artux.pdanetwork.models.user.dto.StoryData;

import java.util.Collection;

public interface ResetService {

    Status sendResetPasswordLetter(String email);

    Status changePassword(String token, String password);

    StoryData resetData();

    Collection<String> getTokens();

}

File: pdanet/src/main/java/net/artux/pdanetwork/service/user/reset/ResetService.kt
package net.artux.pdanetwork.service.user.reset

import net.artux.pdanetwork.models.Status
import net.artux.pdanetwork.models.user.dto.StoryData

interface ResetService {
    fun sendResetPasswordLetter(email: String): Status
    fun changePassword(token: String, password: String): Status
    fun resetData(): StoryData
    fun getTokens(): Collection<String>
    fun deleteAccount(): Boolean
}
